compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/viewinterop/UIKitInteropContainer.uikit.kt

```diff
@@ -92,16 +96,18 @@ internal class UIKitInteropContainer(
         }
 
         val isAdded = interopViews.put(interopView, holder) == null
-
-        val countBelow = countInteropComponentsBelow(holder)
+        val countBelow = countInteropComponentsBelow(holder) {
+            contains(it) && (it as UIKitInteropElementHolder<*>).isOverlay == holder.isOverlay
+        }
+        val container = if (holder.isOverlay) overlayContainer else backgroundContainer
```

Comment:
Please note that properties is public argument in composable function, so it might be changed via recomposition.
In this case, please make sure that you re-add it based on new value